src/features/quality-stats/StatsAggregator.js

```diff
@@ -228,22 +270,30 @@ class StatsAggregator {
 
         // Go through each peer connection and compute aggregates.
         Object.keys(extractedData).forEach(pc => {
-            resultMap[pc] = { isP2P: extractedData[pc].isP2P,
-                usesRelay: extractedData[pc].usesRelay,
-                dtlsErrors: extractedData[pc].dtlsErrors,
-                dtlsFailure: extractedData[pc].dtlsFailure };
-
-            const pcTrackStats = this._calculateTrackStats(extractedData[pc]);
-            const pcTrackResults = this._calculateTrackAggregates(extractedData[pc]);
-            const pcTransportResults = this._calculateTransportAggregates(extractedData[pc]);
+            const pcData = extractedData[pc];
+
+            resultMap[pc] = { isP2P: pcData.isP2P,
+                usesRelay: pcData.usesRelay,
+                dtlsErrors: pcData.dtlsErrors,
+                dtlsFailure: pcData.dtlsFailure };
+
+            const pcResults = resultMap[pc];
+
+            const pcTrackStats = this._calculateTrackStats(pcData);
+            const pcTrackResults = this._calculateTrackAggregates(pcData);
+            const pcTransportResults = this._calculateTransportAggregates(pcData);
             const pcVideoExperienceResults
-                = this._calculateVideoExperienceAggregates(extractedData[pc].inboundVideoExperiences);
+                = this._calculateVideoExperienceAggregates(pcData.inboundVideoExperiences);
+
+            pcResults.iceReconnects = this._calculateReconnects(pcData);
+            pcResults.pcSessionDurationMs = this._calculateSessionDurationMs(pcData);
+            pcResults.iceFailed = this._didIceConnectionFail(pcData);
 
-            resultMap[pc].tracks = pcTrackStats;
-            resultMap[pc].trackAggregates = pcTrackResults;
-            resultMap[pc].transportAggregates = pcTransportResults;
+            pcResults.tracks = pcTrackStats;
+            pcResults.trackAggregates = pcTrackResults;
+            pcResults.transportAggregates = pcTransportResults;
```

Comment:
Wouldn't it make sense to move these three up to where we call the _calculate...() functions, like you do now for duration and the ICE states?

Reply:
yes it would :) , changing.